### ISSUE TITLE
Suddenly broke uploading packages (mupx)

### ISSUE THREAD

**Issue by artpolikarpov** [arunoda/meteor-up-legacy#762]
I use a mupx for a long time without complaints, and suddenly in one of my projects uploading just stopped working.

<img width="1033" alt="screen shot 2015-11-18 at 04 15 00" src="https://cloud.githubusercontent.com/assets/870859/11230334/f7f7c7e8-8dae-11e5-99e0-c7f6e144c231.png">

All commands work — `mupx logs`, `mupx reconfig`. But `mupx deploy` doesn’t! I also can connect to this servers with my ssh-key. And I can deploy sibling-project to sibling-servers (in the same data-center, with the same ssh-key).

I have no idea what to check, because of poor error message. Please help?

**Comment by MasterJames** [comment]
This could also be caused by the mongod.lock file.
Remove /var/lib/mongodb/mongod.lock and see if that helps. In mupx the Mongo container gets jammed by the volume mounted folders lock file. It's meant to preserve your database but this triggers a reboot loop of the docker images for mongodb and that becomes unconnectable so the meteor container also fails.
I made a PR for Mongo Dump and Restore that includes tools to remove the locks. Also note you're suppose to do a manual repair when manually purging locks.

**Comment by artpolikarpov** [comment]
@MasterJames I do not install Mongo on that servers...

**Comment by arunoda** [comment]
Did you see the preloaded bar? If not try changing the network.

**Comment by artpolikarpov** [comment]
> Did you see the preloaded bar? If not try changing the network.

No! But in sibling-config (same ssh-key, same data-center for servers) everything works...

**Comment by arunoda** [comment]
Expose DEBUG=\* to the console and run the deployment. It will give more
info.

**Comment by artpolikarpov** [comment]
Not much: http://d.pr/i/19T8P

It was working a few day ago. Config is the same.

**Comment by artpolikarpov** [comment]
Tried to `mupx setup` again on this servers, and got this:

```
Started TaskList: Setup (linux)
[104.197.83.173] - Installing Docker
[104.197.83.173] - Installing Docker: SUCCESS
[104.197.83.173] - Setting up Environment
[104.197.83.173] x Setting up Environment: FAILED

    -----------------------------------STDERR-----------------------------------
    usermod: /etc/passwd.3071: No space left on device
    usermod: cannot lock /etc/passwd; try again later.
    -----------------------------------STDOUT-----------------------------------
    ----------------------------------------------------------------------------
```

How to fix that? Clean up the logs or what?

**Comment by artpolikarpov** [comment]
The `sudo df -h` shows this:

![](http://d.pr/i/1lyTU+)

**Comment by artpolikarpov** [comment]
It looks like we need to somehow clean up the `/var/lib/docker/aufs/mnt` folder...

**Comment by artpolikarpov** [comment]
After manual cleaning that folder and redeploying the app disk usage lowered to 24%:

![](http://d.pr/i/1464A+)

**Comment by artpolikarpov** [comment]
I am prettry sure it happens because of growing logs:

![](http://d.pr/i/15gPM+)

How can we change this or config?

**Comment by artpolikarpov** [comment]
We can easily fix that by adding log rollover (see https://github.com/docker/docker/pull/11485). As I see it is all about few flags:

```
docker run --logging-driver rollover --log-driver-opts max_size=1k --log-driver-opts file_count=10
```

And also we have to add some new options to mup.json to make it configurable. What you think?

/@arunoda

**Comment by MasterJames** [comment]
Sounds important thanks for the input and discovery efforts.
It's changed slightly in that discussion so in 1.8 it's --log-opt 
https://docs.docker.com/v1.8/reference/logging/overview/

**Comment by artpolikarpov** [comment]
I’m going to make a PR.

**Comment by arunoda** [comment]
Yeah! Send me that.

**Comment by artpolikarpov** [comment]
Wrote and tested simple wrapper over `--log-driver` and `--log-opt` flags for configuring logging drivers.

Options in mup.json looks like this:

## ![](http://d.pr/i/I18l+)

![](http://d.pr/i/1fDFa+)

So now even possilbe to send logs to remote syslog/gelf servers natively!

If you’re ok with that API, I’ll send it to you soon :)

**Comment by arunoda** [comment]
Thanks. In the case of first one, we need to get logs via mup logs command.

Does it work ?

I assume this log-driver features (specially) comes with a later version of
docker. So, we need to check in the deployment stage and warm and abort the
process.

**Comment by artpolikarpov** [comment]
Well, `mup logs` command is available only for the `json-file` logging driver (it is default). It works if you specify `max-size` and `max-file` options. And it do not return anything if you change `log.driver` to `syslog` or whatever. This is the behavior of the Docker.

These features works since Docker v1.8 version, on servers we have 1.9.

**Comment by artpolikarpov** [comment]
I think it is ok to make a remark in docs that `mup logs` will be silent in case of alternative logging drivers. So it will be up to deployers.

**Comment by arunoda** [comment]
Then it's great. We don't need to get it when we are using alternate log
drivers. Yeah! We can do a note.
There are some older mup deployments may contains Docker versions other
than 0.1.8.  Then these options may not work, so in this case we need to
warn the user.

Also, make sure to set of sensible defaults.

**Comment by artpolikarpov** [comment]
I’ll leave defaults as is. If you don’t specify any of log options in the object, they will not be honored, and docker will work as now. I think it is no need to show all of these options in `example/mup.json` because for different drivers you have to specify different options and you can’t mix it.

I’m going to add a short section to a README.md about new possibilities.

About waring user in case of old Docker... We don’t have to do anything because Docker will do it for us: if you specify some unsupported arguments Docker doesn’t start and shows you why.

**Comment by artpolikarpov** [comment]
We also can add an another note in readme that these options works with fresh Docker.

P. S. I’m trying to solve this (not critical) issue as a designer and not like as a coder ;)

**Comment by arunoda** [comment]
Okay. That's okay about the docker client.

But we need to add some reasonable defaults to avoid the issues like you've
faced.
So, adding max-size=100m and max-files=10 seems good  for me.

**Comment by artpolikarpov** [comment]
Aha, I understand now. Ok, I’ll add it.

**Comment by arunoda** [comment]
Great.
